rpki-validator/src/main/java/net/ripe/rpki/validator3/domain/validation/TrustAnchorState.java

```diff
@@ -0,0 +1,46 @@
+package net.ripe.rpki.validator3.domain.validation;
+
+import lombok.extern.slf4j.Slf4j;
+import net.ripe.rpki.validator3.storage.data.TrustAnchor;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * It is to keep track of the state of a TA. A TA transitions to the VALIDATED when
+ * and only when the data for it has been downloaded and the tree validation has been
+ * performed.
```

Comment:
For me it would have been easier to get this if the class doc (or line 23) listed that we lock on the HashMap because HashMaps are unsafe under concurrent modification and that we want to provide allValidated to provide consistent results.

Reply:
Not sure I understand what you mean.
In a nutshell, the whole class is used to switch to UNKNOWN when we got the data but haven't validated it yet and switch to VALIDATED when we finish the validation. The locking stuff is an implementation detail.

Reply:
True, implementation detail that can be figured out. Just takes a bit longer for me to get (and then re-read) upon first read.